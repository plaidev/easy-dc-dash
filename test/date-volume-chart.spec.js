import Vue from 'vue';
import { mount, createLocalVue } from 'vue-test-utils';
import CssModuleTestHelperMixin from './helpers/css-modules-test-helper-mixin';

import Store from '../libs/store';
import DateVolumeChart from '@/date-volume-chart.vue';

const localVue = createLocalVue();
localVue.use(CssModuleTestHelperMixin);

describe('date-volume-chart', () => {
  Store.registerData([
    { date: '2017-01-01T15:00:00.000Z', d1: 'a', d2: true, v: 1 },
    { date: '2017-01-02T15:00:00.000Z', d1: 'a', d2: false, v: 2 },
    { date: '2017-01-03T15:00:00.000Z', d1: 'b', d2: false, v: 3 },
    { date: '2017-01-04T15:00:00.000Z', d1: 'b', d2: true, v: 5 }
  ]);

  const wrapper = mount(localVue.extend(DateVolumeChart), {
    localVue,
    propsData: {
      id: 'test-date-volume-chart',
      dimension: 'd.date',
      reduce: 'd.v'
    },
    attachToDocument: true
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
