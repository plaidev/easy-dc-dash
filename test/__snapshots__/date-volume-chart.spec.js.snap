// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date-volume-chart to match snapshot 1`] = `
<div class="outer-container" style="width: 960px; height: 90px;">
    <div class="backdrop"></div>
    <div class="card-container">
        <div class="inner-container" style="width: 960px; height: 90px;">
            <div class="container-header">
                <div class="icon-box">
                    <i class="fa fa-window-maximize"></i>
                </div>
                <!---->
            </div>
            <div class="render-area">
                <div id="test-date-volume-chart" class="krt-dc-component dc-chart" style="display: flex; position: relative; width: 100%; height: 100%;">
                    <!---->
                    <div class="reset-button">
                        <a class="reset" style="display: none;">
                            <i class="fa fa-ban">&nbsp;</i> <span>フィルター解除</span></a>
                    </div>
                    <!---->
                    <svg width="200" height="200">
                        <g>
                            <g class="grid-line horizontal" transform="translate(40,20)"></g>
                            <g class="grid-line vertical" transform="translate(40,20)">
                                <line x1="0" y1="150" x2="0" y2="0" opacity="0"></line>
                                <line x1="15" y1="150" x2="15" y2="0" opacity="0"></line>
                                <line x1="30" y1="150" x2="30" y2="0" opacity="0"></line>
                                <line x1="45" y1="150" x2="45" y2="0" opacity="0"></line>
                                <line x1="60" y1="150" x2="60" y2="0" opacity="0"></line>
                                <line x1="75" y1="150" x2="75" y2="0" opacity="0"></line>
                                <line x1="90" y1="150" x2="90" y2="0" opacity="0"></line>
                                <line x1="105" y1="150" x2="105" y2="0" opacity="0"></line>
                                <line x1="120" y1="150" x2="120" y2="0" opacity="0"></line>
                            </g>
                            <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-date-volume-chart-clip)">
                                <g class="stack _0">
                                    <rect class="bar" fill="#1f77b4" y="150" height="0"></rect>
                                    <rect class="bar" fill="#1f77b4" y="150" height="0"></rect>
                                    <rect class="bar" fill="#1f77b4" y="150" height="0"></rect>
                                    <rect class="bar" fill="#1f77b4" y="150" height="0"></rect>
                                    <text class="barLabel" text-anchor="middle"></text>
                                    <text class="barLabel" text-anchor="middle"></text>
                                    <text class="barLabel" text-anchor="middle"></text>
                                    <text class="barLabel" text-anchor="middle"></text>
                                </g>
                            </g>
                            <g class="axis x" transform="translate(40,170)">
                                <g class="tick" style="opacity: 0.000001;" transform="translate(0,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Thu 18</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(15,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">12 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(30,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Fri 19</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(45,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">12 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(60,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Sat 20</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(75,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">12 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(90,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">May 21</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(105,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">12 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(120,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Mon 22</text>
                                </g>
                                <path class="domain"></path>
                            </g>
                            <g class="axis y" transform="translate(40,20)">
                                <path class="domain"></path>
                            </g>
                            <g class="brush" transform="translate(40,20)" style="pointer-events: all;">
                                <rect class="background" style="visibility: hidden; cursor: crosshair;" x="0" width="120" height="150"></rect>
                                <rect class="extent" style="cursor: move;" x="0" width="0" height="150"></rect>
                                <g class="resize e" style="cursor: ew-resize; display: none;" transform="translate(0,0)">
                                    <rect x="-3" width="6" height="150" style="visibility: hidden;"></rect>
                                    <path d="M0.5,50A6,6 0 0 1 6.5,56V94A6,6 0 0 1 0.5,100ZM2.5,58V92M4.5,58V92"></path>
                                </g>
                                <g class="resize w" style="cursor: ew-resize; display: none;" transform="translate(0,0)">
                                    <rect x="-3" width="6" height="150" style="visibility: hidden;"></rect>
                                    <path d="M-0.5,50A6,6 0 0 0 -6.5,56V94A6,6 0 0 0 -0.5,100ZM-2.5,58V92M-4.5,58V92"></path>
                                </g>
                            </g>
                        </g>
                        <defs>
                            <clippath id="test-date-volume-chart-clip">
                                <rect width="120" height="150" transform="translate(-0, -0)"></rect>
                            </clippath>
                        </defs>
                    </svg>
                </div>
            </div>
        </div>
    </div>
</div>
`;
